dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -895,50 +895,61 @@ void SchemaBuilder<Getter, NameMapper>::syncAllSchema()
     /// Create all databases.
     std::vector<DBInfoPtr> all_schemas = getter.listDBs();
 
-    // TODO:make parallel to speed up
     std::unordered_set<String> db_set;
 
-    for (const auto & db : all_schemas)
-    {
-        std::shared_lock<std::shared_mutex> shared_lock(shared_mutex_for_databases);
-        if (databases.find(db->id) == databases.end())
-        {
-            shared_lock.unlock();
-            applyCreateSchema(db);
-            db_set.emplace(name_mapper.mapDatabaseName(*db));
-            LOG_DEBUG(log, "Database {} created during sync all schemas", name_mapper.debugDatabaseName(*db));
-        }
-    }
+    //We can't use too large default_num_threads, otherwise, the lock grabbing time will be too much.
+    size_t default_num_threads = std::max(4UL, std::thread::hardware_concurrency());
+    auto sync_all_schema_thread_pool = ThreadPool(default_num_threads, default_num_threads / 2, default_num_threads * 2);
+    auto sync_all_schema_wait_group = sync_all_schema_thread_pool.waitGroup();
 
-    // TODO:make parallel to speed up
+    std::mutex set_mutex;
     for (const auto & db : all_schemas)
     {
-        std::vector<TableInfoPtr> tables = getter.listTables(db->id);
-        for (auto & table : tables)
-        {
-            LOG_INFO(log, "Table {} syncing during sync all schemas", name_mapper.debugCanonicalName(*db, *table));
-
-            /// Ignore view and sequence.
-            if (table->is_view || table->is_sequence)
+        auto task = [this, &db, &db_set, &set_mutex] {
```

Comment:
```suggestion
        auto task = [this, =db, &db_set, &set_mutex] {
```
Should we use `=db` instead of `&db` to copy the ptr?